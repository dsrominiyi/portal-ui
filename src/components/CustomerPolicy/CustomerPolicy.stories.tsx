import React from 'react';
import { storiesOf } from '@storybook/react';
import CustomerPolicy from '.';

const stories = storiesOf('CustomerPolicy', module);

const validPolicy: Policy = {
  id: '12347',
  type: 'pc',
  insurer: 'Bob Smith Ltd',
  startDate: '2019-11-01',
  endDate: '2020-11-01',
  rate: 345.89
};

const invalidPolicy: Policy = {
  id: '12345',
  type: 'invalid',
  insurer: 'We Insure Any Car',
  startDate: '2019-01-01',
  endDate: '2020-01-01'
};

const render = (policy: Policy) => () => <CustomerPolicy policy={policy} />;

const lapsedPolicy: Policy = {
  id: '12347',
  type: 'pc',
  insurer: 'Bob Smith Ltd',
  startDate: '2018-11-01',
  endDate: '2019-11-01'
};

export const storyRenders = {
  validPolicy: render(validPolicy),
  invalidPolicy: render(invalidPolicy),
  lapsedPolicy: render(lapsedPolicy)
};

stories.add('Valid policy', render(validPolicy));
stories.add('Invalid policy', render(invalidPolicy));
stories.add('Lapsed policy', render(lapsedPolicy));
