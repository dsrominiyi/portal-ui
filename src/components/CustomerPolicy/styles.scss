.time-remaining-container {
  width: 80px;
}
