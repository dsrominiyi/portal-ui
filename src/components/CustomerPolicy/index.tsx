import React, { FunctionComponent } from 'react';

import PolicyWidget from '../PolicyWidget';
import TimeRemaining from '../TimeRemaining';
import policyTypeMapper from '../../helpers/policyTypeMapper';

import styles from './styles.scss';

interface Props {
  policy: Policy;
}

export const CustomerPolicy: FunctionComponent<Props> = ({ policy }: Props) => {
  const type = policyTypeMapper(policy.type);

  if (!type) {
    return null;
  }

  const additionalItems = [
    <div className={styles.timeRemainingContainer} key={policy.id}>
      <TimeRemaining startDate={policy.startDate} endDate={policy.endDate} />
    </div>
  ];

  return <PolicyWidget id={policy.id} type={type} additionalItems={additionalItems} />;
};

CustomerPolicy.displayName = 'CustomerPolicy';
export default CustomerPolicy;
