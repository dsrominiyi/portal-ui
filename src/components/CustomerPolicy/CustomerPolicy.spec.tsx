import { storyRenders } from './CustomerPolicy.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('CustomerPolicy', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
