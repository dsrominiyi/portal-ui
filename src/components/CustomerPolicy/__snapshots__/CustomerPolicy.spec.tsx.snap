// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerPolicy component story renders match snapshots: invalidPolicy 1`] = `""`;

exports[`CustomerPolicy component story renders match snapshots: lapsedPolicy 1`] = `
<PolicyWidget
  additionalItems={
    Array [
      <div
        className="timeRemainingContainer"
      >
        <TimeRemaining
          endDate="2019-11-01"
          startDate="2018-11-01"
        />
      </div>,
    ]
  }
  id="12347"
  type="pc"
/>
`;

exports[`CustomerPolicy component story renders match snapshots: validPolicy 1`] = `
<PolicyWidget
  additionalItems={
    Array [
      <div
        className="timeRemainingContainer"
      >
        <TimeRemaining
          endDate="2020-11-01"
          startDate="2019-11-01"
        />
      </div>,
    ]
  }
  id="12347"
  type="pc"
/>
`;
